src/scripts/components/LoginForm.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+
+import { Button } from '../@toggl/ui/components';
+import Spinner from './Spinner';
+
+interface LoginFormProps {
+  onSubmit: (email: string, password: string) => Promise<ButtonResponse | null>;
+  onSuccess: () => void;
+  onError: (error: string) => void;
+}
+
+export default function LoginForm ({ onSubmit, onSuccess, onError }: LoginFormProps) {
+  const [loading, setLoading] = React.useState(false);
+  const emailRef = React.useRef<HTMLInputElement>(null);
+  const passwordRef = React.useRef<HTMLInputElement>(null);
+
+  const handleSubmit = React.useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      const email = emailRef.current && emailRef.current.value;
+      const password = passwordRef.current && passwordRef.current.value;
+
+      if (!email || !password) return;
+
+      setLoading(true);
+      const response = await onSubmit(email, password);
+      setLoading(false);
+
+      if (!response) return;
```

Comment:
Just curious face: shouldn't we handle it at least somehow?

Reply:
Good catch, it's an impossible case in this flow, but still added `Unkown Error` state in accordance with the LoginPage's login handler